### PR TITLE
Steps

### PR DESCRIPTION
**Pull request by artvalborgcp**

### CODE REVIEW

File: terraform/instance.tf [hunk 1]
```diff
@@ -0,0 +1,42 @@
+resource "google_compute_instance_template" "tpl" {
+  name         = "template"
+  machine_type = "f1-micro"
```
Comment (yevheniimykhailin) [inline terraform/instance.tf]: variable

File: terraform/instance.tf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  disk {
+    source_image = "debian-cloud/debian-10"
+    auto_delete  = true
+    disk_size_gb = 10
+    boot         = true
```
Comment (yevheniimykhailin) [inline terraform/instance.tf]: variables

File: terraform/instance.tf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    email  = google_service_account.sa.email
+    scopes = ["cloud-platform"]
+  }
+
+}
+
+resource "google_compute_instance_from_template" "tpl" {
+  name = "bookshelfinstance1"
+  zone = "us-central1-a"
```
Comment (yevheniimykhailin) [inline terraform/instance.tf]: variable

File: terraform/instance.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  network_interface {
+    network = "mygcpnet"
+#    subnetwork = "mygcpsubnet"
+  }
+
+  metadata = {
+    foo = "bar"
+  }
```
Comment (yevheniimykhailin) [inline terraform/instance.tf]: if is not in use - delete

File: terraform/instance.tf [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+}
+
+resource "google_compute_instance_from_template" "tpl" {
+  name = "bookshelfinstance1"
+  zone = "us-central1-a"
+
+  source_instance_template = google_compute_instance_template.tpl.id
+
+  can_ip_forward = false
```
Comment (yevheniimykhailin) [inline terraform/instance.tf]: why do you need ip forward?

File: terraform/instance.tf [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  labels = {
+    my_key = "my_value"
+  }
+}
```
Comment (yevheniimykhailin) [inline terraform/instance.tf]: labels = var.labels, use variables

File: terraform/mynetwork.tf [hunk 7]
```diff
@@ -0,0 +1,52 @@
+resource "google_compute_network" "mygcpnet" {
+name                    = "mygcpnet"
+#auto_create_subnetworks = "true"
```
Comment (yevheniimykhailin) [inline terraform/mynetwork.tf]: set to false, or custom mode. be explicit.

File: terraform/mynetwork.tf [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+resource "google_compute_subnetwork" "mygcpsubnet" {
+  name          = "mygcpsubnet"
+  network       = google_compute_network.mygcpnet.id
+  ip_cidr_range = "10.0.0.0/24"
+  region        = "us-central1"
```
Comment (yevheniimykhailin) [inline terraform/mynetwork.tf]: variables

File: terraform/mynetwork.tf [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+
+
+resource "google_compute_firewall" "mygcpnet-allow-http-ssh-rdp-icmp" {
+name = "mygcpnet-allow-http-ssh-rdp-icmp"
+network = google_compute_network.mygcpnet.self_link
+allow {
+    protocol = "tcp"
+    ports    = ["22", "80", "3389"]
```
Comment (yevheniimykhailin) [inline terraform/mynetwork.tf]: 3389?

File: terraform/mynetwork.tf [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  name                               = "mygcpnat"
+  router                             = google_compute_router.mygcprouter.name
+  region                             = google_compute_router.mygcprouter.region
+  nat_ip_allocate_option             = "AUTO_ONLY"
+  source_subnetwork_ip_ranges_to_nat = "ALL_SUBNETWORKS_ALL_IP_RANGES"
+}
+
+
+resource "google_service_account" "sa" {
+  account_id   = "my-gcp1"
```
Comment (yevheniimykhailin) [inline terraform/mynetwork.tf]: Names should be no more than 30, easy to understand, according to naming conventions. For example, sa-TEAM-PURPOSE.

File: terraform/outputs.tf [hunk 11]
```diff
@@ -0,0 +1,14 @@
+output "bookshelfinstance_ip" {
```
Comment (yevheniimykhailin) [inline terraform/outputs.tf]: each output has to have description field aligned with value.

File: terraform/provider.tf [hunk 12]
```diff
@@ -0,0 +1,3 @@
+provider "google" {
```
Comment (yevheniimykhailin) [inline terraform/provider.tf]: correct with TF 13+ style.
for example:
`terraform {  
  required_version = ">= 0.13"  
  required_providers {    
    google = {      
       source  = "hashicorp/google"      
       version = ">= 3.43, <4.0"    
}
}`

File: terraform/sql.tf [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+      private_network = google_compute_network.mygcpnet.id
+    }
+  }
+}
+
+provider "google-beta" {
+  region  = "us-central1"
+  zone    = "us-central1-a"
+  project = "my-gcp-terraform"
+}
```
Comment (yevheniimykhailin) [inline terraform/sql.tf]: Use TF 13+ style.

File: 7-gce/config.py [hunk 14]
```diff
@@ -106,5 +106,5 @@
 # If you receive a invalid redirect URI error review you settings to ensure
 # that the current URI is allowed.
 GOOGLE_OAUTH2_CLIENT_ID = \
-    'your-client-id'
-GOOGLE_OAUTH2_CLIENT_SECRET = 'your-client-secret'
+    ''
```
Comment (yevheniimykhailin) [inline 7-gce/config.py]: ?

**Review by yevheniimykhailin — CHANGES_REQUESTED** [review]

File: 7-gce/config.py [hunk 15]
```diff
@@ -106,5 +106,5 @@
 # If you receive a invalid redirect URI error review you settings to ensure
 # that the current URI is allowed.
 GOOGLE_OAUTH2_CLIENT_ID = \
-    '540860343698-of5tonk1rc7vjfgq1o6lq1vcb2oc1hsv.apps.googleusercontent.com'
-GOOGLE_OAUTH2_CLIENT_SECRET = '-OgeC9nWCSor0rEwx8B9p-hO'
+    ''
```
Comment (yevheniimykhailin) [inline 7-gce/config.py]: https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository

**Review by yevheniimykhailin — CHANGES_REQUESTED** [review]
Using docs remove creds from git permanently.

File: 7-gce/config.py (hunk 15 above)
Comment (artvalborgcp) [reply 7-gce/config.py]: solved

**Comment by artvalborgcp** [comment]
w

**Pull request closed without merging**